### PR TITLE
os/arch/xtensa: add missing dep-path for xtensa

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
1. esp32/spi_flash dir is missed in makefile for making depend
2. remove useless code for nonexist dir 'common'

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [3c491d7c0570d211837303342ca140a9cb809610] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3c491d7c0570d211837303342ca140a9cb809610] - Code Rule Check (C++) OK.

File: os/arch/xtensa/src/Makefile [hunk 1]
```diff
@@ -69,21 +69,18 @@ ifeq ($(CONFIG_WINDOWS_NATIVE),y)
   ARCH_SRCDIR = $(TOPDIR)\arch\$(CONFIG_ARCH)\src
   TINYARA = "$(TOPDIR)\$(OUTBIN_DIR)\tinyara$(EXEEXT)"
   CFLAGS += -I$(ARCH_SRCDIR)\chip
-  CFLAGS += -I$(ARCH_SRCDIR)\common
```
Comment (ChinaTizenRT) [inline os/arch/xtensa/src/Makefile]: for the code, it looks good to me,  
for commit log, it seems lack of removing unused "common"  path.

Comment (sunghan-chang) [reply os/arch/xtensa/src/Makefile]: I agree. But I am sorry I already merged it.

**Pull request merged**